src/ussd_registration.js

```diff
@@ -263,25 +265,28 @@ go.app = function() {
                     return go.utils
                         .get_identity_by_address({'msisdn': msisdn}, self.im)
                         .then(function(contact) {
-                            if (    contact && contact.details && contact.details.addresses &&
-                                    contact.details.addresses.msisdn && contact.details.addresses.msisdn[msisdn] &&
-                                    contact.details.addresses.msisdn[msisdn].optedout) {
-                                self.im.user.set_answer('mother_id', contact.id);
-                                self.im.user.set_answer('receiver_id', contact.id);
-                                if (bypassPostbirth) {
-                                    self.im.user.set_answer('state_pregnancy_status', 'prebirth');
-                                    return self.states.create('state_last_period_month');
-                                } else {
-                                    return self.states.create('state_pregnancy_status');
-                                }
-                            }
-                            else if (contact && contact.details && contact.details.receiver_role) {
-                                self.im.user.set_answer('role_player', contact.details.receiver_role);
-                                self.im.user.set_answer('contact_id', contact.id);
-                                return 'state_msisdn_already_registered';
-                            } else {
+                            if (contact == undefined) {
```

Comment:
We probably want to check the rest of the code for this too.

Reply:
@rudigiesler I'm not convinced the behaviour above isn't what we want. If there isn't a contact for this number then, regardless of how it's represented, we want to send them down this branch of logic.
Are there circumstances under which the contact could be `none` but that identity actually exists?

Reply:
I'm not sure what the javascript client returns under different circumstances, but I think this would fall under "doing something fancy", in which case I would prefer if we were more explicit about it, ie `if (contact === undefined || contact === null)`